### PR TITLE
Simplify class weighting code

### PR DESCRIPTION
**Pull request by davegrays**
eliminates duplicated code

### CODE REVIEW

File: ultralytics/cfg/default.yaml [hunk 1]
```diff
@@ -100,6 +99,7 @@ cls: 0.5  # (float) cls loss gain (scale with pixels)
 dfl: 1.5  # (float) dfl loss gain
 pose: 12.0  # (float) pose loss gain
 kobj: 1.0  # (float) keypoint obj loss gain
+cls_weights: # (str | list[float], optional) class weights applied to cls loss (object detection)
```
Comment (zshureih) [inline ultralytics/cfg/default.yaml:102]: nice change

File: ultralytics/engine/trainer.py [hunk 2]
```diff
@@ -239,16 +239,16 @@ def calc_weights(self):
             print("Not using class weights")
         elif isinstance(cls_weights, list) and len(cls_weights) == nc:
             cls_weights = torch.Tensor(cls_weights)
-        elif cls_weights == "median":
-            cls_weights = self.get_median_frequency_weights(nc)
+        elif cls_weights == "median-inverse":
```
Comment (zshureih) [inline ultralytics/engine/trainer.py]: this feels wordy

File: ultralytics/engine/trainer.py [hunk 3]
```diff
@@ -239,16 +239,16 @@ def calc_weights(self):
...
+            cls_weights = self.get_inverse_class_frequency_weights(nc, median_norm=True)
         elif cls_weights == "inverse": 
             cls_weights = self.get_inverse_class_frequency_weights(nc)
         else:
             raise ValueError("Invalid value for cls_weights. Please use None, a list of length num_classes, 'median', or 'inverse'.")
         
         return cls_weights
 
-    def get_inverse_class_frequency_weights(self, nc):
+    def get_inverse_class_frequency_weights(self, nc, median_norm=False):
```
Comment (zshureih) [inline ultralytics/engine/trainer.py]: Good change!

**Review by zshureih — APPROVED** [review]

File: ultralytics/engine/trainer.py [hunk 4]
```diff
@@ -239,16 +239,16 @@ def calc_weights(self):
...
         elif isinstance(cls_weights, list) and len(cls_weights) == nc:
             cls_weights = torch.Tensor(cls_weights)
-        elif cls_weights == "median":
-            cls_weights = self.get_median_frequency_weights(nc)
+        elif cls_weights == "median-inverse":
+            cls_weights = self.get_inverse_class_frequency_weights(nc, median_norm=True)
         elif cls_weights == "inverse": 
             cls_weights = self.get_inverse_class_frequency_weights(nc)
         else:
             raise ValueError("Invalid value for cls_weights. Please use None, a list of length num_classes, 'median', or 'inverse'.")
```
Comment (zshureih) [inline ultralytics/engine/trainer.py]: Please update the options given by the error message. 

ie 'inverse' to 'median-inverse' if that's the option we end up going with

**Review by zshureih — CHANGES_REQUESTED** [review]

File: ultralytics/engine/trainer.py (hunk 2 above)
Comment (davegrays) [reply ultralytics/engine/trainer.py]: fair enough, I just want to indicate that the two methods are really identical except for the numerator , i.e. median-count vs total-count / nc

File: ultralytics/engine/trainer.py (hunk 4 above)
Comment (davegrays) [reply ultralytics/engine/trainer.py]: done, lots of new options now, check it out

File: ultralytics/engine/trainer.py (hunk 2 above)
Comment (davegrays) [reply ultralytics/engine/trainer.py]: check out the new options

**Review by zshureih — APPROVED** [review]
LG2M!

**Pull request merged**